subsys/net/ip/tcp2.c

```diff
@@ -291,6 +467,11 @@ static void tcp_send_process(struct k_work *work)
 
 		pkt = forget ? tcp_slist(&conn->send_queue, get, struct net_pkt,
 						next) : tcp_pkt_clone(pkt);
+		if (!pkt) {
```

Comment:
It's OK to handle the clone error, but in order not to introduce another error, perhaps should be handled like in lines 460-461.

In principal, should we send a RST before? But since the clone fails, will the allocation for RST succeed?

Reply:
> It's OK to handle the clone error, but in order not to introduce another error, perhaps should be handled like in lines 460-461.

So if the clone fails, we terminate the connection? If we really run out of mem, then that is probably ok. We could try to send RST but as you said, it might fail too.